registry/remote/example_test.go

```diff
@@ -0,0 +1,624 @@
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// application/vnd.oci.image.manifest.v1+json
+	// sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b
+	// 24
+}
+
+// ExampleRepository_Fetch_byTag gives example snippets for downloading a manifest by tag.
+func ExampleRepository_Fetch_manifestByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag)
+	if err != nil {
+		panic(err)
+	}
+	rc, err := repo.Fetch(ctx, descriptor)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err)
+	}
+
+	fmt.Println(string(pulledBlob))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_Fetch_manifestByDigest gives example snippets for downloading a manifest by digest.
+func ExampleRepository_Fetch_manifestByDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	// resolve the blob descriptor to obtain the size of the blob
+	descriptor, err := repo.Resolve(ctx, digest)
+	if err != nil {
+		panic(err)
+	}
+	rc, err := repo.Fetch(ctx, descriptor)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulled, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulled) || descriptor.Size != int64(len(pulled)) {
+		panic(err)
+	}
+
+	fmt.Println(string(pulled))
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_manifestByTag gives example snippets for downloading a manifest by tag with only one API call.
+func ExampleRepository_FetchReference_manifestByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	tag := "latest"
+	// specify the expected manifest media type for the blob
+	reg.ManifestMediaTypes = append(reg.ManifestMediaTypes, ocispec.MediaTypeImageManifest)
+	descriptor, rc, err := repo.FetchReference(ctx, tag)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err)
+	}
```

Comment:
Looks like we have lots of duplicated code like this.

@Wwwsylvia What do you think if we promote [internal/ioutil.ReadAll()](https://github.com/oras-project/oras-go/blob/main/internal/ioutil/io.go#L37) to the `content` package?

Reply:
That would be useful.

Reply:
Issue created: #128